kartothek/io/dask/repartition.py

```diff
@@ -0,0 +1,94 @@
+import math
+from typing import Optional
+
+import dask
+import dask.dataframe as dd
+
+import kartothek.io.dask.dataframe as ktk_ddf
+from kartothek.core.factory import DatasetFactory
+from kartothek.serialization import PredicatesType
+
+
+def optimize(ddf: dd.DataFrame) -> dd.DataFrame:
+    """
+    Optimize a dask graph using the dask.dataframe optimizer.
+    Parameters
+    ----------
+    ddf
+        Dask Dataframe
+    Returns
+    --------
+    dask.DataFrame
+    """
+    return dask.optimize(ddf, optimizations=[dask.dataframe.optimize])[0]
+
+
+def calculate_repartition_ratio(
+    dataset_factory: DatasetFactory,
+    table_name: str,
+    predicates: Optional[PredicatesType] = None,
+    *,
+    target_nrows_per_partition: int,
+    dataset_metadata_frac: float,
+) -> dd.core.Scalar:
+    """
+    Calculate a `repartition_ratio` according to the dataset input and `target_nrows_per_partition` (the desired
+    number of rows of each dataframe partition).
+
+    Parameters
+    ----------
+    dataset_factory
+        dataset factory
+    table_name
+        Dataset Table name
+    predicates
+        Optional , but predicates to be applied during calculatin repartition ratio
+    target_nrows_per_partition
+        target num of rows per partition
+    dataset_metadata_frac
+        used during collection of dataset metadata
+    Returns
```

Comment:
The section requires a newline to separate it from the above section. I believe otherwise the docs are not rendered properly, are they?

Reply:
Agreed. Updated the doc section accordingly.